src/aws-cpp-sdk-core/source/config/EndpointResolver.cpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/core/config/EndpointResolver.h>
+#include <aws/core/utils/StringUtils.h>
+#include <aws/core/config/ConfigAndCredentialsCacheManager.h>
+#include <aws/core/platform/Environment.h>
+#include <aws/core/utils/logging/LogMacros.h>
+
+namespace Aws
+{
+    namespace Config
+    {
+        namespace EndpointResolver
+        {
+            Aws::String ToEnvSuffix(const Aws::String& serviceId)
+            {
+                Aws::String result = Utils::StringUtils::ToUpper(serviceId.c_str());
+                Utils::StringUtils::Replace(result, " ", "_");
+                Utils::StringUtils::Replace(result, "-", "_");
+                return result;
+            }
+
+            Aws::String EndpointSource(const Aws::String& serviceId, const Aws::String& profileName)
+            {
+                const Aws::String serviceKey = ToEnvSuffix(serviceId);
+
+                // 1) Check ignore flag from environment variable
+                Aws::String ignoreEnv = Aws::Environment::GetEnv("AWS_IGNORE_CONFIGURED_ENDPOINT_URLS");
+                if (!ignoreEnv.empty() && Utils::StringUtils::ConvertToBool(ignoreEnv.c_str())) {
+                    AWS_LOGSTREAM_DEBUG(ENDPOINT_RESOLVER_TAG, "Configured endpoints ignored due to AWS_IGNORE_CONFIGURED_ENDPOINT_URLS=true");
+                    return "";
+                }
+
+                // 1b) Check ignore flag from profile (early check)
+                if (!profileName.empty() && HasCachedConfigProfile(profileName)) {
+                    Profile profile = GetCachedConfigProfile(profileName);
+                    const Aws::String ignoreVal = profile.GetValue("ignore_configured_endpoint_urls");
+                    if (!ignoreVal.empty() && Utils::StringUtils::ConvertToBool(ignoreVal.c_str())) {
+                        AWS_LOGSTREAM_DEBUG(ENDPOINT_RESOLVER_TAG, "Configured endpoints ignored due to ignore_configured_endpoint_urls=true in profile: " << profileName);
+                        return "";
+                    }
+                }
+
+                // 2) Service-specific environment variable
+                {
+                    Aws::String service = "AWS_ENDPOINT_URL_";
+                    service += serviceKey;
+                    Aws::String fromEnv = Aws::Environment::GetEnv(service.c_str());
+                    if (!fromEnv.empty()) {
+                        AWS_LOGSTREAM_DEBUG(ENDPOINT_RESOLVER_TAG, "Resolved configured endpoint from service-specific environment variable: " << service);
```

Comment:
why are we using DEBUG and TRACE in the same if statement, that means that no matter which log level you choose one or both will be displayed. additionally when you want to know one, but not the other?